### ISSUE TITLE
related fields default value

### ISSUE THREAD

**Issue by Mitschmaster** [odoo/odoo#49230]
## Impacted versions:
13.0

## Steps to reproduce:
class Mother(models.Model):
_name = 'mother'
line = fields.one2many('son', 'mother_id')
currency_id = fields.many2one('res.currency', default=lambda self: self.env.company.currency_id.id)

class Son(models.Model):
_name= 'son'
mother_id = fields.many2one('mother)
currency_id = fields.many2one(related='mother_id.currency_id')
somefield = fields.Monetary()

## Current behavior:
when adding a line to mother, the currency_id of son is not set. if it is changed in mother, the change also goes to son.

## Expected behavior:
son should recieve the default value when created.

**Comment by pedrobaeza** [comment]
Not really. You should set the default as well on the children, or wait until it's synchronized on save. Other option is to switch to computed fields.

**Comment by Mitschmaster** [comment]
can i set a default value to a related field?
i already tried that and it led to an infinite recursion error. but maybe i did something other wrong.

**Comment by pedrobaeza** [comment]
Not sure either right now. Switch to computed then.

**Comment by Mitschmaster** [comment]
i did not have luck with either default or computed. an onchange('line') in the mother did work now.

**Comment by pedrobaeza** [comment]
Uhm, not sure then if it's a side effect. @rco-odoo can you please check to confirm?
